frontend/src/components/FilterPills.js

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faTimesCircle } from '@fortawesome/free-solid-svg-icons';
+import moment from 'moment';
+import { formatDateRange } from './DateRangeSelect';
+
+const filterProp = PropTypes.shape({
+  topic: PropTypes.string,
+  condition: PropTypes.string,
+  query: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)]),
+  id: PropTypes.string,
+  toggleAllChecked: PropTypes.bool,
+});
+
+/* Pill */
+export function Pill({ filter, isFirst, onRemoveFilter }) {
+  const {
+    id,
+    topic,
+    condition,
+    query,
+    toggleAllChecked,
+  } = filter;
+
+  const filterNameLookup = [
+    {
+      topic: 'role',
+      display: 'Specialist',
+    },
+    {
+      topic: 'startDate',
+      display: 'Date Range',
+    },
+  ];
+
+  const determineFilterName = () => {
+    const topicMatch = filterNameLookup.find((f) => f.topic === topic);
+
+    if (topicMatch) {
+      return topicMatch.display;
+    }
+
+    return topic;
+  };
+
+  const determineQuery = (keepOriginalLength = true) => {
```

Comment:
This method gets pretty long once we start adding additional topics to the filter. I wonder if we can keep a list of topics and the type of query and just handle each query by type (date, multi-select, string, etc...). We already have the start of that list up in the `filterNameLookup` variable.

Reply:
I have updated the code to have an array of objects that we add to in the future for new filter items.